src/Form/Form.js

```diff
@@ -3,27 +3,54 @@ import classes from "./Form.module.css";
 import * as firebase from "../lib/firebase";
 
 const Form = props => {
+  const { setdbItems, dbItems, token, onEnterToken } = props;
   const [name, setName] = useState("");
   const [frequency, setFrequency] = useState(0);
   const [date, setDate] = useState("");
 
   const handleSubmit = e => {
     e.preventDefault();
     let db = firebase.fb.firestore();
-    let data = {
-      name,
-      frequency,
-      date
-    };
-    db.collection(props.token)
-      .add(data)
-      .then(function() {
-        alert("Item added!");
-      })
-      .catch(function(error) {
-        alert("Something went wrong!");
-        console.error("Error writing document: ", error);
-      });
+    let newName = name
+      .toUpperCase()
+      .replace(/[^\w\s]|_/g, "")
+      .replace(/\s+/g, " ");
+    const existingItem = dbItems.find(element => {
+      let newElement = element.name
+        .toUpperCase()
+        .replace(/[^\w\s]|_/g, "")
+        .replace(/\s+/g, " ");
+      return newElement === newName;
+    });
+
+    if (!existingItem) {
```

Comment:
I believe that instead of manually searching for the item with `dbItems.find`, you can query the database [with their API](https://firebase.google.com/docs/firestore/query-data/queries). This may save you some work if you choose to go down this route:
```js
function checkIfListHasElement(name) {
  const listRef = db.collection(token);
  const query = listRef.where("name", "==", name);
  return query.then(querySnapshot => querySnapshot.length > 0);
}
```

Reply:
I only used .where() in C#. I didn't know it was available in javascript too. Thanks!